### PR TITLE
fix: updated isLoaded and isReady conditions for mixpanel

### PR DESCRIPTION
**Pull request by sandeepdsvs**
## PR Description

Resolves INT-1852
Updated isLoaded and isReady conditions for mixpanel

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved the accuracy of the `isLoaded` and `isReady` checks in the Mixpanel integration by updating the property used for verification.
- **Tests**
	- Updated tests for `isLoaded` and `isReady` functionality in the Mixpanel integration to include asynchronous behavior.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js [hunk 1]
```diff
@@ -98,11 +98,11 @@ class Mixpanel {
 
   isLoaded() {
     window.mixpanel.register({ mp_lib: 'Rudderstack: web' });
-    return !!window?.mixpanel?.config;
+    return !!window?.mixpanel?.__loaded;
   }
 
   isReady() {
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:104]: If there is no change in the logic, let's simply invoke `isLoaded` here.

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js [hunk 2]
```diff
@@ -98,11 +98,11 @@ class Mixpanel {
 
   isLoaded() {
     window.mixpanel.register({ mp_lib: 'Rudderstack: web' });
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:100]: `isLoaded` is potentially invoked multiple times during the initialization process. So, we shouldn't register here. Please move it to the `init` method.

Also, we should change `Rudderstack: web` to `RudderStack Web` if there is no breaking change.

File: __tests__/prodsdk.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
...)}catch(e){}return t}(e);i.ajs_aid&&Iu.toBeProcessedArray.push(["setAnonymousId",i.ajs_aid]),i.ajs_uid&&Iu.toBeProcessedArray.push(["identify",i.ajs_uid,r(i,t)]),i.ajs_event&&Iu.toBeProcessedArray.push(["track",i.ajs_event,r(i,n)])}(window.location.href),Eu&&Au.forEach((function(e){return Iu.toBeProcessedArray.push(e)})),ku&&ku.length>0&&(ku.shift(),Iu.load.apply(Iu,c(ku)));var Su=Iu.ready.bind(Iu),Tu=Iu.identify.bind(Iu),wu=Iu.page.bind(Iu),Ou=Iu.track.bind(Iu),_u=Iu.alias.bind(Iu),Ru=Iu.group.bind(Iu),Pu=Iu.reset.bind(Iu),Du=Iu.load.bind(Iu),Lu=Iu.initialized=!0,xu=Iu.getUserId.bind(Iu),Mu=Iu.getSessionId.bind(Iu),Bu=Iu.getUserTraits.bind(Iu),Nu=Iu.getAnonymousId.bind(Iu),Fu=Iu.setAnonymousId.bind(Iu),ju=Iu.getGroupId.bind(Iu),Uu=Iu.getGroupTraits.bind(Iu),Gu=Iu.startSession.bind(Iu),Ku=Iu.endSession.bind(Iu),Hu=Iu.setAuthToken.bind(Iu);return e.alias=_u,e.endSession=Ku,e.getAnonymousId=Nu,e.getGroupId=ju,e.getGroupTraits=Uu,e.getSessionId=Mu,e.getUserId=xu,e.getUserTraits=Bu,e.group=Ru,e.identify=Tu,e.initialized=Lu,e.load=Du,e.page=wu,e.ready=Su,e.reset=Pu,e.setAnonymousId=Fu,e.setAuthToken=Hu,e.startSession=Gu,e.track=Ou,Object.defineProperty(e,"__esModule",{value:!0}),e}({});
```
Comment (saikumarrs) [inline __tests__/prodsdk.js:1]: This file should not be committed. Revert it please.

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js [hunk 4]
```diff
@@ -98,11 +98,11 @@ class Mixpanel {
 
   isLoaded() {
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:99]: Can we add unit tests for both the updated functions? It's mandatory now.

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js [hunk 5]
```diff
@@ -98,11 +98,11 @@ class Mixpanel {
...
     window.mixpanel.register({ mp_lib: 'Rudderstack: web' });
-    return !!window?.mixpanel?.config;
+    return !!window?.mixpanel?.__loaded;
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:101]:
```suggestion
    return !!window.mixpanel?.__loaded;
```

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 2 above)
Comment (saikumarrs) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:100]: `mp_lib` is a super property. Let's not change it now; it'll be a breaking change.

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 5 above)
Comment (sandeepdsvs) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js]: Done

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 1 above)
Comment (sandeepdsvs) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:108]: Done

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 4 above)
Comment (sandeepdsvs) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js:104]: Done

File: packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js (hunk 2 above)
Comment (sandeepdsvs) [reply packages/analytics-js-integrations/src/integrations/Mixpanel/browser.js]: Done. Now registering in init.

**Review by yashasvibajpai — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**